### ISSUE TITLE
infur schema by looking at the structure as you go

### ISSUE THREAD

**Issue by dominictarr** [mafintosh/turbo-json-parse#16]
okay, okay, I'm trolling

but why is this so fast? does this mean that 1/2 the ops are actually type checks? But isn't JSON.parse written in C? hmm, code is here: https://github.com/v8/v8/blob/b8626ca445554b8376b5a01f651b70cb8c01b7dd/src/json-parser.cc

**Comment by dominictarr** [comment]
oh I see! We have data that partially stable (outer bit is stable, then content field is arbitary object) @arj03 never the less had some success with adapting your turbo-json-parse module. @arj03 did you see jitson?

**Comment by mafintosh** [comment]
@dominictarr are the ordering of your keys stable as well? that gives an additional good boost with jitson. We should implement the `any` type in JSON schema to support that (I think that's what @arj03 is already doing also).

**Comment by dominictarr** [comment]
mostly. we recently locked this down, there are two acceptable orders on the top level. https://github.com/ssbc/ssb-validate/pull/10 order is also significant - signatures won't validate if the order isn't preserved.

**Comment by mafintosh** [comment]
nice. the parser supports both but consistent order is def faster to parse

**Comment by arj03** [comment]
Yeah, it was a bit strange seeing this project. We found out I think half a year ago that json parse was really slow. Including the binary part which this project also support, so it was like a very nicely packaged gift in the middle of August. Santa comes early :-)

Thank you!

**Comment by dominictarr** [comment]
@mafintosh another idea, have been wondering about and in-place json - instead of parsing, just read out a value at a path. i.e. scan to that path, then parse that one value, don't create objects.

my binary in-place format was way faster than json, the implementation isn't optimized though, just the design. [some results here](https://github.com/dominictarr/bench-flumelog-offset)
